### PR TITLE
T&P: Some tech debt work on tests

### PR DESCRIPTION
**Pull request by kicferk1**
## Ticket and context

I got paranoid about some edge cases and while I was at it, decided to do some tech debt on our unit tests.

### Is there anything that the code reviewer should know?

I might do the same for participant actions, there is a lot of duplication and possibly missing small edge cases there too. 

### Code quality checks
- [x] All commit messages are meaningful and true
- [x] Added enough automated tests

### CODE REVIEW

File: app/services/record_declarations/base.rb [hunk 1]
```diff
@@ -21,8 +21,8 @@ class Base
     delegate :schedule, :participant_declarations, to: :user_profile, allow_nil: true
 
     class << self
-      def call(params)
-        new(params).call
+      def call(params:)
+        new(params: params).call
```
Comment (plisovin) [inline app/services/record_declarations/base.rb:25]: What is the benefit to have named parameter here?

Comment (kicferk1) [reply app/services/record_declarations/base.rb:25]: Consistency with the participants service.

And the benefit of that is... we can have shared examples between them to show the behaviour for "when user is not a participant" or "when user does not have access to the course".

It's the difference of `described_class.call(params)` vs `described_class.call(params: params)` in tests.

And, well, I opted for making it a named parameter rather than making participant actions one an un-named parameter because I thought that would lead to fewer changes.

File: spec/support/shared_examples/participant_declaration_support.rb [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+      travel_to cutoff_end_datetime + 1.day
+    end
+
+    it "raises ParameterMissing error" do
+      params = given_params.merge({ declaration_date: (cutoff_end_datetime + 1.day).rfc3339 })
+      expect { described_class.call(params: params) }.to raise_error(ActionController::ParameterMissing)
+    end
+  end
+
+  it "fails when user profile is a withdrawn record" do
```
Comment (plisovin) [inline spec/support/shared_examples/participant_declaration_support.rb:123]: Another use case is ```deferred``` profile

Comment (kicferk1) [reply spec/support/shared_examples/participant_declaration_support.rb:123]: Good spot :)

**Review by plisovin — APPROVED** [review]
